Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js

```diff
@@ -1135,4 +1343,232 @@ describe('gdjs.PlatformerObjectRuntimeBehavior', function () {
   });
+
+  describe('(walk)', function () {
+    let runtimeScene;
+    let object;
+    let platform;
+
+    beforeEach(function () {
+      runtimeScene = makeTestRuntimeScene();
+
+      // Put a platformer object on a platform
+      object = new gdjs.RuntimeObject(runtimeScene, {
+        name: 'obj1',
+        type: '',
+        behaviors: [
+          {
+            type: 'PlatformBehavior::PlatformerObjectBehavior',
+            name: 'auto1',
+            gravity: 1500,
+            maxFallingSpeed: 1500,
+            acceleration: 500,
+            deceleration: 1500,
+            maxSpeed: 500,
+            jumpSpeed: 900,
+            canGrabPlatforms: true,
+            ignoreDefaultControls: true,
+            slopeMaxAngle: 60,
+            jumpSustainTime: 0.2,
+            roundCoordinates: true,
+          },
+        ],
+      });
+      object.getWidth = function () {
+        return 10;
+      };
+      object.getHeight = function () {
+        return 20;
+      };
+      runtimeScene.addObject(object);
+
+      // Put a platform.
+      platform = addPlatformObject(runtimeScene);
+      platform.setPosition(0, -10);
+    });
+
+    const fall = (frameCount) => {
+      for (let i = 0; i < frameCount; ++i) {
+        const lastY = object.getY();
+        runtimeScene.renderAndStep(1000 / 60);
+        expect(object.getBehavior('auto1').isFalling()).to.be(true);
+        expect(object.getBehavior('auto1').isMoving()).to.be(true);
+        expect(object.getY()).to.be.above(lastY);
+      }
+    };
+
+    const walkRight = (frameCount) => {
+      for (let i = 0; i < frameCount; ++i) {
+        const lastX = object.getX();
+        object.getBehavior('auto1').simulateRightKey();
+        runtimeScene.renderAndStep(1000 / 60);
+        expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+        expect(object.getX()).to.be.above(lastX);
+        //TODO probably a bug, uncomment when it's fixed.
+        //expect(object.getBehavior('auto1').isMoving()).to.be(true);
+      }
+    };
+
+    const fallOnPlatform = (maxFrameCount) => {
+      // Ensure the object falls on the platform
+      for (let i = 0; i < maxFrameCount; ++i) {
+        runtimeScene.renderAndStep(1000 / 60);
+      }
+      //Check the object is on the platform
+      expect(object.getY()).to.be(-30); // -30 = -10 (platform y) + -20 (object height)
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+    };
+
+    it('can walk from a platform to another one', function () {
+      const platform2 = addPlatformObject(runtimeScene);
+      platform2.setPosition(platform.getX() + platform.getWidth(), platform.getY());
+
+      object.setPosition(30, -32);
+      // Ensure the object falls on the platform
+      fallOnPlatform(10);
+
+      // walk from the 1st platform to the 2nd one
+      walkRight(30);
+      expect(object.getX()).to.be.above(platform2.getX());
+    });
```

Comment:
maybe check if it's on floor too?

Reply:
walkRight do the checks too. The name doesn't reflect it, but I think it make the test easier to read to only have a description of what the player do.

Reply:
@D8H what do you think of checking that too?

Reply:
I added a check on the Y position. Was it what you meant?

Reply:
I was thinking of checking if `isOnFloor` was true, but maybe not necessary!

Reply:
Ok, walkRight() does it.

Reply:
Perfect then, thanks!